Crash when getting URI MIME type.

#### Software and hardware versions 
Collect v1.9.1

#### Problem description
Crash in `FormEntryActivity` when getting intent URI MIME type.

#### Steps to reproduce the problem
Unknown, from Play Store [here](https://play.google.com/apps/publish/?dev_acc=00659255839475882647#AndroidMetricsErrorsPlace:p=org.odk.collect.android&appVersion=PRODUCTION&clusterName=apps/org.odk.collect.android/clusters/9956b8d6) (requires credentials). It only affects Android 6.0 but on a range of devices.

The line that crashes is `String uriMimeType = getContentResolver().getType(uri);`. Either the content resolver can be `null` or `getType` dereferences `uri`?

#### Expected behavior
No crash.

#### Other information 
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2572)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2654)
  at android.app.ActivityThread.-wrap11 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1488)
  at android.os.Handler.dispatchMessage (Handler.java:111)
  at android.os.Looper.loop (Looper.java:207)
  at android.app.ActivityThread.main (ActivityThread.java:5728)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:789)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:679)
Caused by: java.lang.NullPointerException: 
  at com.android.internal.util.Preconditions.checkNotNull (Preconditions.java:60)
  at android.content.ContentResolver.getType (ContentResolver.java:336)
  at org.odk.collect.android.activities.FormEntryActivity.onCreate (FormEntryActivity.java:367)
  at android.app.Activity.performCreate (Activity.java:6360)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1113)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2519)
```

--- a/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
@@ -367,7 +367,7 @@ public class FormEntryActivity extends AppCompatActivity implements AnimationLis
                 Uri uri = intent.getData();
                 String uriMimeType = null;
 
-                if(uri!=null) {
+                if(uri != null) {
                     uriMimeType = getContentResolver().getType(uri);
                 }
 

--- a/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
@@ -367,7 +367,7 @@ public class FormEntryActivity extends AppCompatActivity implements AnimationLis
                 Uri uri = intent.getData();
                 String uriMimeType = null;
 
-                if(uri != null) {
+                if (uri != null) {
                     uriMimeType = getContentResolver().getType(uri);
                 }
 

--- a/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
@@ -365,16 +365,10 @@ public class FormEntryActivity extends AppCompatActivity implements AnimationLis
             Intent intent = getIntent();
             if (intent != null) {
                 Uri uri = intent.getData();
-                String uriMimeType =null;
-                
-                //Check if uri is not null
+                String uriMimeType = null;
+
                 if(uri!=null) {
                     uriMimeType = getContentResolver().getType(uri);
-                }
-                else
-                {
-                    //uri is null. So, toast an error.
-                    showCustomToast(getString(R.string.error_occured), Toast.LENGTH_SHORT);
                 }
 
                 if (uriMimeType != null && uriMimeType.equals(InstanceColumns.CONTENT_ITEM_TYPE)) {

--- a/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
@@ -366,6 +366,7 @@ public class FormEntryActivity extends AppCompatActivity implements AnimationLis
             if (intent != null) {
                 Uri uri = intent.getData();
                 String uriMimeType =null;
+                
                 //Check if uri is not null
                 if(uri!=null) {
                     uriMimeType = getContentResolver().getType(uri);

--- a/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
@@ -365,7 +365,16 @@ public class FormEntryActivity extends AppCompatActivity implements AnimationLis
             Intent intent = getIntent();
             if (intent != null) {
                 Uri uri = intent.getData();
-                String uriMimeType = getContentResolver().getType(uri);
+                String uriMimeType =null;
+                //Check if uri is not null
+                if(uri!=null) {
+                    uriMimeType = getContentResolver().getType(uri);
+                }
+                else
+                {
+                    //uri is null. So, toast an error.
+                    showCustomToast(getString(R.string.error_occured), Toast.LENGTH_SHORT);
+                }
 
                 if (uriMimeType != null && uriMimeType.equals(InstanceColumns.CONTENT_ITEM_TYPE)) {
                     // get the formId and version for this instance...

